dataproc/dataproc.go

```diff
@@ -0,0 +1,304 @@
+	zoneID      = flag.String("zone", "", "Cloud Platform zone")
+)
+
+// clusterConfig defines the confuration of a cluster including its project, name, and region.
+type clusterConfig struct{
+		bucket,
+    project,
+		region,
+		state,
+		name,
+		uuid,
+		zone string
+}
+
+func main() {
+	// Parse command line arguments
+	flag.Parse()
+
+	// Check for required flags (golang flags does not support?)
+	if *bucketName == "" || *clusterName == "" || *projectID == "" || *pysparkFile == "" || *region == "" || *zoneID == "" {
+		log.Fatal("Incorrect arguments specified see 'go-dataproc -help' for help")
+	}
+
+	// Create a new Cloud Dataproc service
+	ctx := context.Background()
+	client, err := google.DefaultClient(ctx, scope)
+	if err != nil {
+		log.Fatal(err)
+	}
+	service, err := dataproc.New(client)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Create a new Google Clout Storage service
+	storageService, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Create a new clusterConfig to hold details about this cluster
+	cluster := clusterConfig{project: *projectID, region: *region, name: *clusterName, zone: *zoneID}
+
+	// Create a cluster
+	if _, err := createCluster(service, &cluster); err != nil {
+		log.Fatal(err)
+	}
+	log.Println("Cluster created")
+
+	// Wait on the cluster to become active
+	log.Println("Waiting for cluster to be ready")
+	if _, err = waitForCluster(service, &cluster); err != nil {
+		log.Fatal(err)
+	}
+
+	// Submit a job to the cluster
+	log.Println("Submitting job")
+	jobID, err := submitJob(service, storageService, *pysparkFile, *bucketName, &cluster)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Wait for the job to complete
+	log.Println("Waiting for job to complete")
+	if _, err = waitForJob(service, jobID, &cluster); err != nil {
+		log.Fatal(err)
+	}
+	log.Println("Job is finished")
+
+	output, err := getJobOutput(storageService, jobID, &cluster)
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Printf("Job output:\n%s\n", output)
+
+	// Delete the cluster
+	if _, err = deleteCluster(service, &cluster); err != nil {
+		log.Fatal(err)
+	}
+	log.Println("Cluster deleted")
+}
+
+// createCluster creates a Cloud Dataproc cluster with the given name and region.
+func createCluster(service *dataproc.Service, cluster *clusterConfig) (response string, err error) {
+	// Create a gceConfig object for the cluster
+	gceConfig := dataproc.GceClusterConfig{
+		ZoneUri: fmt.Sprintf(computeUriBase, cluster.project, cluster.zone),
+	}
+
+	// Create a (Dataproc API) clusterConfig for the cluster
+	clusterConfig := dataproc.ClusterConfig{
+		GceClusterConfig: &gceConfig,
+	}
+
+	// Create a cluster object
+	clusterSpec := dataproc.Cluster{
+		ClusterName: cluster.name,
+		ProjectId:   cluster.project,
+		Config:      &clusterConfig,
+	}
+
+	// Create the cluster
+	res, err := service.Projects.Regions.Clusters.Create(cluster.project, cluster.region, &clusterSpec).Do()
+
+	return fmt.Sprintf("%s", res), err
```

Comment:
Just `return ...Do()`, perhaps? The formatted string is not used except for tests.